### PR TITLE
Dashboard filter

### LINKED ISSUE #759: Public Dashboard Filters
### What ###
A button "use Filters for Public Dasboard" on the Shifts Page.
The button generates a link similar to the /user_shifts parameters for the dashboard (omitting time).
The Dashboard shows open shifts and statistics only for the selected rooms and/or angeltypes.

### Why ###
Teams might use the dashboard filtering only their vacancies. This can be locally (like a display in Hall 1 showing all open Hall 1 Shifts or the local Hackspace at rC3), by angeltype/team (e.g. CERT Medical, CERT Technical, CERT Dispatcher), or even by shifttype (show all open Access Control shifts).

### PR DESCRIPTION
**Pull request by MyIgel**
This adds a "Filter" link to the dashboard which filters the shifts using the users filter from the shifts list. Also "Dashboard" link is added to Rooms and Angeltypes to the corresponding dashboards

This also fixes an exception when editing an Angeltype while session expires

closes #759 (Public Dashboard Filters)

### CODE REVIEW

File: includes/view/AngelTypes_view.php [hunk 1]
```diff
@@ -358,7 +358,10 @@ function AngelType_view(
                 $angeltype,
                 $shiftsFilterRenderer,
                 $shiftCalendarRenderer
-            )
+            ),
+            __('Dashboard') => [
```
Comment (msquare) [inline includes/view/AngelTypes_view.php:362]: Why use a tab as link? I would expect a link to another part of the app to be in the top button row.
Please add the dashboard icon.

File: includes/view/Rooms_view.php [hunk 2]
```diff
@@ -45,6 +45,8 @@ function Room_view(Room $room, ShiftsFilterRenderer $shiftsFilterRenderer, Shift
         $shiftCalendarRenderer->render()
     ]);
 
+    $tabs[__('Dashboard')] = ['href' => url('/public-dashboard', ['filtered' => true, 'rooms' => [$room->id]])];
```
Comment (msquare) [inline includes/view/Rooms_view.php:48]: See AngelTypes.

**Review by msquare — CHANGES_REQUESTED** [review]
Nice! Only minor change requests, as always.

File: includes/view/AngelTypes_view.php (hunk 1 above)
Comment (MyIgel) [reply includes/view/AngelTypes_view.php:362]: Maybe add it to the date select, hmm

**Pull request merged**